pkg/resources/manager/manager.go

```diff
@@ -193,6 +151,10 @@ func (r *resourceManager) initialize(ctx context.Context, machines []*api.Machin
 	r.mx.Lock()
 	defer r.mx.Unlock()
 
+	if len(r.sources) == 0 {
```

Comment:
cannot we move it into initialize function? do we want to support resource manager without resources?